### PR TITLE
Remove appointment, encounter, medication dispense

### PR DESCRIPTION
**Pull request by ilaflamme-va**
I removed a lot of utility code around these resources as well. 
Thoughts?

### CODE REVIEW

**Comment by jblefkove-va** [comment]
Should the mappings be removed from `DataQueryIdsCodebookSupplier` ?

**Comment by jblefkove-va** [comment]
These look like cruft:

```
./data-query/src/main/java/gov/va/api/health/dataquery/service/config/MagicReferenceConfig.java: 
<li>Automatically filter out references for resources that are optional, such as Appointment

./data-query/src/main/java/gov/va/api/health/dataquery/service/controller/organization/Dstu2OrganizationController.java: 
Request Mappings for DSTU Organization, see https://www.hl7.org/fhir/DSTU2/appointment.html for

./data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/crawler/ReferenceInterceptorTest.java: 
"https://example.com/api/Appointment/615f31df-f0c7-5100-ac42-7fb952c630d0")
"https://example.com/api/Appointment/615f31df-f0c7-5100-ac42-7fb952c630d0");

./data-query/src/test/java/gov/va/api/health/dataquery/service/config/Dstu2JacksonMapperTest.java:
 ReferenceSerializerProperties disableEncounter =
 new MagicReferenceConfig("https://example.com", "dstu2", "stu3", disableEncounter))
 ReferenceSerializerProperties disableEncounter =
 "https://example.com", "api/dstu2", "api/stu3", disableEncounter))
 ReferenceSerializerProperties disableEncounter =
 "https://example.com", "api/dstu2", "api/stu3", disableEncounter))

./data-query/src/test/java/gov/va/api/health/dataquery/service/config/Stu3JacksonMapperTest.java:
 ReferenceSerializerProperties disableEncounter =
 new MagicReferenceConfig("https://example.com", "dstu2", "stu3", disableEncounter))
 ReferenceSerializerProperties disableEncounter =
 "https://example.com", "api/dstu2", "api/stu3", disableEncounter))
 ReferenceSerializerProperties disableEncounter =
 "https://example.com", "api/dstu2", "api/stu3", disableEncounter))

./data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/crawler/ConcurrentResourceBalancingRequestQueueTest.java:
 "foo/api/Encounter/2",
 "foo/api/Encounter/1",
 "foo/api/Encounter/1",
 "foo/api/Encounter/2",
 q.add("foo/api/Encounter/3");
 assertThat(q.next()).isEqualTo("foo/api/Encounter/3");
```

File: data-query/src/main/java/gov/va/api/health/dataquery/service/config/ReferenceSerializerProperties.java [hunk 1]
```diff
@@ -22,35 +22,24 @@
...
   private boolean practitioner;
 
-  private boolean medicationDispense;
-
   /**
    * Return boolean property from property file for the listed resources. Return true by default.
    */
   public boolean checkForResource(String resourceName) {
     switch (resourceName) {
-      case "Appointment":
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/config/ReferenceSerializerProperties.java]: Location, Organization, and Practitioner are all available in datamart now. Should we just drop this class? We aren't worried about references anymore.

**Comment by jblefkove-va** [comment]
PlantUML diagrams should also be updated

File: data-query/src/main/resources/application.properties [hunk 2]
```diff
@@ -36,8 +37,6 @@ conformance.security.token-endpoint=unset
 conformance.security.authorize-endpoint=unset
 conformance.security.description=http://docs.smarthealthit.org/
 conformance.resourceDocumentation=Implemented per specification. See http://hl7.org/fhir/DSTU2/http.html
-included-references.appointment=true
```
[the remark this answers is not in the document]
Comment (ilaflamme-va) [reply data-query/src/main/resources/application.properties]: It does allow us to feature toggle. Whether it's necessary... /shrug

**Pull request merged**